### ISSUE TITLE
explore marisa-trie in place of our trie implementation 

### ISSUE THREAD

**Issue by gregcaporaso** [biocore/scikit-bio#937]
This may be important for future OTU picking in QIIME 2. [marisa-trie](https://pypi.python.org/pypi/marisa-trie).

**Comment by wasade** [comment]
:+1:

**Comment by jairideout** [comment]
I think the next step is to deprecate the trie code in scikit-bio in favor of marisa-trie or one of the other trie Python packages (e.g., [datrie](https://pypi.python.org/pypi/datrie/), [DAWG](http://dawg.readthedocs.org/en/latest/)). Each trie implementation has its pros and cons (e.g., memory vs. runtime efficiency, immutable vs. mutable, etc.). The functionality provided by scikit-bio's trie object can be accomplished using one of these other packages.

**Comment by gregcaporaso** [comment]
I agree re: deprecation of our trie code in favor of one of the others,
though we'll just need to make sure that we can get the information we need
from those (i.e., the sets of ids that cluster to each node in the trie).
We may then want to wrap that code in our sequence clustering API (which is
yet to be defined, but which we'll need as core functionality in skbio for
QIIME 2).

**Comment by jairideout** [comment]
> though we'll just need to make sure that we can get the information we need from those

Sounds good, are you interested in taking the lead on this and posting back here when it's safe to deprecate this object?

**Comment by colinbrislawn** [comment]
Here is another trie implementation, which may be useful.
https://github.com/dcjones/hat-trie

It's even got a python wrapper:
https://github.com/kmike/hat-trie 

One comment on the page:
>If you need a memory efficient data structure and don't need inserts then marisa-trie or DAWG should work better.

So if we DO need inserts, then this looks like a solid option.

**Comment by gregcaporaso** [comment]
Yep, I'll take this one.
